x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -636,206 +401,6 @@ const AssistantComponent: React.FC<Props> = ({
     [assistantTelemetry, currentConversation?.title]
   );
 
-  const refetchConversationsState = useCallback(async () => {
-    await refetchResults();
-  }, [refetchResults]);
-
-  const queryClient = useQueryClient();
-
-  const { mutateAsync } = useMutation<Conversation | undefined, unknown, Conversation>(
-    ['SET_DEFAULT_CONNECTOR'],
-    {
-      mutationFn: async (payload) => {
-        const apiConfig = getGenAiConfig(defaultConnector);
-        return setApiConfig({
-          conversation: payload,
-          apiConfig: {
-            ...payload?.apiConfig,
-            connectorId: (defaultConnector?.id as string) ?? '',
-            actionTypeId: (defaultConnector?.actionTypeId as string) ?? '.gen-ai',
-            provider: apiConfig?.apiProvider,
-            model: apiConfig?.defaultModel,
-            defaultSystemPromptId: allSystemPrompts.find((sp) => sp.isNewConversationDefault)?.id,
-          },
-        });
-      },
-      onSuccess: async (data) => {
-        await queryClient.cancelQueries({ queryKey: CONVERSATIONS_QUERY_KEYS });
-        if (data) {
-          queryClient.setQueryData<{ data: Conversation[] }>(CONVERSATIONS_QUERY_KEYS, (prev) => ({
-            ...(prev ?? {}),
-            data: uniqBy([data, ...(prev?.data ?? [])], 'id'),
-          }));
-        }
-        return data;
-      },
-    }
-  );
-
-  useEffect(() => {
```

Comment:
this `useEffect` has been eliminated entirely. Now there is a function, `initializeDefaultConversationWithConnector` that handles when a default conversation without a connector is initialized with the default connector. If necessary, `initializeDefaultConversationWithConnector` is called on conversation change/update.